cli/src/config.rs

```diff
@@ -1305,59 +1344,149 @@ mod tests {
                 jj_config: Some("custom.toml".into()),
                 ..Default::default()
             },
-            want: Want::New("custom.toml"),
+            wants: vec![Want::New("custom.toml")],
+        }
+        .run()
+    }
+
+    #[test]
+    fn test_config_path_jj_config_existing_multiple() -> anyhow::Result<()> {
+        TestCase {
+            files: vec!["custom1.toml", "custom2.toml"],
+            env: UnresolvedConfigEnv {
+                jj_config: Some(["custom1.toml", "custom2.toml"].join(PATH_SEPARATOR)),
+                ..Default::default()
+            },
+            wants: vec![
+                Want::Existing("custom1.toml"),
+                Want::Existing("custom2.toml"),
+            ],
+        }
+        .run()
+    }
+
+    #[test]
+    fn test_config_path_jj_config_new_multiple() -> anyhow::Result<()> {
+        TestCase {
+            files: vec!["custom1.toml"],
+            env: UnresolvedConfigEnv {
+                jj_config: Some(["custom1.toml", "custom2.toml"].join(PATH_SEPARATOR)),
+                ..Default::default()
+            },
+            wants: vec![Want::Existing("custom1.toml"), Want::New("custom2.toml")],
         }
         .run()
     }
 
     #[test]
-    fn test_config_path_config_pick_config_dir() -> anyhow::Result<()> {
+    fn test_config_path_jj_config_empty_paths_filtered() -> anyhow::Result<()> {
+        TestCase {
+            files: vec!["custom1.toml"],
+            env: UnresolvedConfigEnv {
+                jj_config: Some(["custom1.toml", "", "custom2.toml"].join(PATH_SEPARATOR)),
+                ..Default::default()
+            },
+            wants: vec![Want::Existing("custom1.toml"), Want::New("custom2.toml")],
+        }
+        .run()
+    }
```

Comment:
nit: can you also add test for `jj_config: Some("".to_owned())`? It should effectively disable config loading.